components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/EditorMicroservice.java

```diff
@@ -844,6 +848,53 @@ public Response getAttributes(@PathParam("siddhiAppName") String siddhiAppName,
                 ).build();
     }
 
+    /**
+     * Converts a given Siddhi App string to a specific JSON format for a graph that diagrammatically
+     * display's the Siddhi App to be generated in the Editor design view.
+     *
+     * @param siddhiAppBase64 The Siddhi App (encoded to Base64) to be converted to JSON
+     * @return The JSON result in a predefined format
+     */
+    @POST
+    @Path("/event-flow")
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response constructEventFlowJsonString(String siddhiAppBase64) {
+        Response response;
+
+        try {
+            String siddhiAppString = new String(Base64.getDecoder().decode(siddhiAppBase64),
+                    Charset.defaultCharset());
+
+            SiddhiAppMap siddhiAppMap = new SiddhiAppMap(siddhiAppString);
+            EventFlow eventFlow = new EventFlow(siddhiAppMap);
+
+            // The 'Access-Control-Allow-Origin' header must be set to '*' as this might be accessed
+            // by other domains in the future.
+            response = Response.status(Response.Status.OK)
+                    .header("Access-Control-Allow-Origin", "*")
+                    .entity(eventFlow.getEventFlowJSON().toString())
+                    .build();
+        } catch (SiddhiAppCreationException e) {
+            // SiddhiAppCreationException is caught if the provided 'SiddhiAppString' contains any syntax errors.
+            log.info("Cannot Generate Graph View: " + e.getMessage());
```

Comment:
Why INFO log?

Reply:
Hi, It's put as an info log because it's not really meant to be displayed as an error in the editor. If the user tries to switch to the design view when the source view has any errors, the following warning message shows up:

![error-demo](https://user-images.githubusercontent.com/25219255/36959904-8e29a420-206a-11e8-9c1a-e0683129db1d.gif)

The log was initially not implemented at all and only a response with a `BAD_REQUEST` was returned. The log was then later added if the user wanted a little more info as to **why the design view cannot be rendered**  and not whether the siddhi app contains any errors or not. That is why I put it as an INFO log and not an ERROR log. Here is the log for the above example:

![error2](https://user-images.githubusercontent.com/25219255/36961414-3b866a62-2071-11e8-96ba-502d0ecb8f46.gif)

Maybe we can change this to an error, but IMHO I do not think the stack trace should be displayed cause the user may think that we are trying to run the app when switching to the design view. But if you still think that this should be changed then I'll change the log into a proper error message.

Reply:
As you explained if the `SiddhiAppCreationException` is not an error in the Editor, rather user have entered an invalid Siddhi app, then lelt's make this to a DEBUG log. In that approach, for us the support developers who might need to know the actual root cause, by enabling DEBUG logs we can get that information.

Reply:
Hi, so I spoke to Damith regarding the matter, and he said that it's ok if we print this as an error with the stack trace as well. Cause anyway the end goal of that line is to inform the user that he/she cannot switch to the graph view because the written siddhi app contains errors.

Reply:
The new implementation would then look like this:

![error3](https://user-images.githubusercontent.com/25219255/36966868-2dc6bf72-2084-11e8-9813-e6efa3303f6d.gif)